src/maintainer/infrastructure.rst

```diff
@@ -294,22 +297,66 @@ principles.
 * We upgrade them in an ad-hoc manner on a periodic basis as we have the time and energy to do so.
   Note that because of the way we enforce runtime constraints, these compiler upgrades will not break
   existing packages. However, if you are using the compilers outside of ``conda``, then you may find issues.
-* We generally provide notice in the form of an announcement when a compiler is going to be upgraded.
-  Note that these changes take a bit of time to complete, so you will generally have time
-  to prepare should you need to.
```

Comment:
Why?

Reply:
Are you asking why I removed this?

Because AFAIK the last gcc & clang updates didn't come with an announcement, and that in turn because it's no as big of a deal anymore as changing to comp7 (which is noted further below in that document as "our current compiler stack") was once upon a time.

Nowadays, gcc & clang bumps just percolate through the ecosystem at the speed with which feedstocks get rerendered, without compatibility constraints (well, aside from small issues here and there, but not large-scale ABI incompatibilties).

Reply:
This is not about ABI compatibilites and if you look at the git history it is not about gcc 4->7 migration.

> aside from small issues here and there, but not large-scale ABI incompatibilties

Exactly why announcements are there

Reply:
> Exactly why announcements are there

If someone had asked me for an announcement for https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4215, I would have done it, no problem. However, neither were [clang 13](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2802), [clang 14](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3261) or [GCC 11](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/3731) announced; also it's the first time during all those bumps (from what I saw) that someone notices a breakage at all.

> This is not about ABI compatibilites and if you look at the git history it is not about gcc 4->7 migration.

OK, I'll readd the announcement bit. However the "Note that these changes take a bit of time to complete, so you will generally have time to prepare should you need to." is not true anymore AFAICT.